internal/pkg/queue/enqueue.go

```diff
@@ -160,32 +170,42 @@ func (q *PersistentGroupedQueue) Enqueue(item *Item) error {
 		return ErrQueueClosed
 	}
 
-	q.mutex.Lock()
-	defer q.mutex.Unlock()
+	var commit uint64 = 0
 
-	// Find free position
-	startPos, err := q.queueFile.Seek(0, io.SeekEnd)
-	if err != nil {
-		return fmt.Errorf("failed to seek to end of file: %w", err)
-	}
+	err := func() error {
```

Comment:
what's the point of using this syntax? you can just return on error of any of the function calls, the mutex will still be deferred

Reply:
![image](https://github.com/user-attachments/assets/8bc66eb8-c93f-4340-847b-3d911fdacde2)

I want to ensure that the execution of this closure (yellow) is locked, but the subsequent green area does not need to be locked.

Reply:
Ok got it, you might wanna declare a function outside the scope of EnqueueUntilCommitted() for clarity of code and debugging. Unnamed functions/routines are not recommended.